packages/swapper/src/swappers/ButterSwap/xhr.ts

```diff
@@ -0,0 +1,210 @@
+  if (result.isErr()) return Err(result.unwrapErr())
+  // No runtime validation, just return the data
+  return Ok(result.unwrap().data)
+}
+
+/**
+ * @see https://docs.butternetwork.io/butter-swap-integration/butter-api-for-routing/get-findtoken
+ */
+export const findToken = async (
+  chainId: number,
+  address: string,
+): ButterSwapPromise<FindTokenResponse> => {
+  const result = await butterService.get<FindTokenResponse>('/findToken', {
+    params: { chainId, address },
+  })
+  if (result.isErr()) return Err(result.unwrapErr())
+  return Ok(result.unwrap().data)
+}
+
+/**
+ * @see https://docs.butternetwork.io/butter-swap-integration/butter-api-for-routing/get-route
+ */
+export interface GetButterRouteArgs {
+  sellAsset: Asset
+  buyAsset: Asset
+  amountHumanUnits: string
+  slippage: string
+  affiliate?: string
+}
+
+const SOLANA_NATIVE_ADDRESS = 'So11111111111111111111111111111111111111112'
+
+export const getButterRoute = async ({
+  sellAsset,
+  buyAsset,
+  amountHumanUnits,
+  slippage,
+  affiliate,
+}: GetButterRouteArgs): ButterSwapPromise<RouteResponse> => {
+  const butterFromChainId = chainIdToButterSwapChainId(sellAsset.chainId)
+  const butterToChainId = chainIdToButterSwapChainId(buyAsset.chainId)
+  if (!butterFromChainId || !butterToChainId) {
+    return Err(
+      makeSwapErrorRight({
+        message: '[getButterRoute] Unsupported chainId',
+        code: TradeQuoteError.UnsupportedChain,
+      }),
+    )
+  }
+
+  const feeAssetId = chainIdToFeeAssetId(sellAsset.chainId)
+  const sellAssetIsNative = sellAsset.assetId === feeAssetId
+  const buyAssetIsNative = buyAsset.assetId === chainIdToFeeAssetId(buyAsset.chainId)
+
+  const { assetReference: sellAssetAddressRaw } = fromAssetId(sellAsset.assetId)
+  const { assetReference: buyAssetAddressRaw } = fromAssetId(buyAsset.assetId)
+
+  const sellAssetAddress = (() => {
+    if (sellAsset.chainId === solanaChainId && sellAssetIsNative) return SOLANA_NATIVE_ADDRESS
+    if (sellAssetIsNative) return zeroAddress
+    return sellAssetAddressRaw
+  })()
+  const buyAssetAddress = (() => {
+    if (buyAsset.chainId === solanaChainId && buyAssetIsNative) return SOLANA_NATIVE_ADDRESS
+    if (buyAssetIsNative) return zeroAddress
+    return buyAssetAddressRaw
+  })()
+  const params = {
+    fromChainId: butterFromChainId,
+    tokenInAddress: sellAssetAddress,
+    toChainId: butterToChainId,
+    tokenOutAddress: buyAssetAddress,
+    amount: amountHumanUnits,
+    type: 'exactIn',
+    slippage,
+    entrance: 'Butter+',
+    affiliate,
+  }
+  const result = await butterService.get<RouteResponse>('/route', { params })
+  if (result.isErr()) return Err(result.unwrapErr())
+  const data = result.unwrap().data
+  return Ok(data)
+}
+
+export type ButterBuildTxArgs = {
+  hash: string
+  slippage: string
+  from: string
+  receiver: string
+}
+
+export const getBuildTx = async ({
```

Comment:
naming: this should probably be `fetchTxData` (i.e I'd never know what `getBuildTx()` is unless I open the def)

Reply:
https://github.com/shapeshift/web/pull/9799/commits/a61b97fff8a12c1b4db9faf5e9b1d570bd2d2eec

Reply:
Same fetch vernacular for other methods? pls 🙏🏽